File: packages/sheets/ios/RNToolsSheetsModule.swift
import ExpoModulesCore

struct SheetAppearance: Record {
    @Field
    var grabberVisible: Bool?

    @Field
    var backgroundColor: String?

    @Field
    var cornerRadius: Float?
}

public class RNToolsSheetsModule: Module {
    public func definition() -> ModuleDefinition {
        Name("RNToolsSheets")

        View(RNToolsSheetsView.self) {
            Events("onDismiss", "onStateChange", "onDismissPrevented")

            Prop("snapPoints") { (view, snapPoints: [CGFloat]) in
                view.updateSnapPoints(snapPoints)
            }

            Prop("isOpen") { (view, isOpen: Bool) in
                view.updateIsOpen(isOpen)
            }

            Prop("initialIndex") { (view, initialIndex: Int) in
                view.updateInitialIndex(initialIndex)
            }

            Prop("appearanceIOS") { (view, appearance: SheetAppearance) in
                view.updateAppearance(
                    grabberVisible: appearance.grabberVisible ?? true,
                    backgroundColor: appearance.backgroundColor,
                    cornerRadius: appearance.cornerRadius
                )
            }

            Prop("canDismiss") { (view, canDismiss: Bool) in
                view.updateCanDismiss(canDismiss)
            }
        }
    }
}

File: packages/sheets/android/src/main/java/expo/modules/sheets/RNToolsSheetsModule.kt
package expo.modules.sheets

import android.view.View
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class RNToolsSheetsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("RNToolsSheets")

    View(RNToolsSheetsView::class) {
      GroupView<RNToolsSheetsView> {
        AddChildView { parent, child: View, index ->
          parent.rootViewGroup.addView(child, index)
        }

        GetChildCount { parent ->
          return@GetChildCount parent.rootViewGroup.childCount
        }

        GetChildViewAt { parent, index ->
          parent.rootViewGroup.getChildAt(index)
        }

        RemoveChildView { parent, child: View ->
          parent.rootViewGroup.removeView(child)
        }

        RemoveChildViewAt { parent, index ->
          parent.rootViewGroup.removeViewAt(index)
        }
      }

      Events("onDismiss", "onStateChange")

      Prop("isOpen") { view: RNToolsSheetsView, isOpen: Boolean ->
        view.props.isOpen = isOpen
      }

      Prop("openToIndex") { view: RNToolsSheetsView, openToIndex: Int ->
        view.props.openToIndex = openToIndex
      }

      Prop("snapPoints") { view: RNToolsSheetsView, snapPoints: List<Int> ->
        view.props.snapPoints = snapPoints.map { view.convertToPx(it) }
      }

      Prop("appearanceAndroid") { view: RNToolsSheetsView, appearance: SheetAppearance ->
        view.props.dimAmount = appearance.dimAmount ?: 0.56f
        view.props.backgroundColor = appearance.backgroundColor
        view.props.cornerRadius = appearance.cornerRadius?.toFloat()
      }
    }
  }
}
